microsoft-azure-storage/src/com/microsoft/azure/storage/core/BaseRequest.java

```diff
@@ -229,8 +235,11 @@ protected PasswordAuthentication getPasswordAuthentication() {
         If we are using https, check if we should enable socket keep-alive timeouts to work around JVM bug.
          */
         if (retConnection instanceof HttpsURLConnection && !options.disableHttpsSocketKeepAlive()) {
+            if(staticKeepAliveSocketFactory == null) {
```

Comment:
This code would need to be thread-safe here.

Reply:
I removed this altogether from `BaseRequest` since this logic is part of the singleton pattern in 'KeepAliveSocketFactory'.